### ISSUE TITLE
Rename main version branch from `3.0` to `3.x`

### ISSUE THREAD

**Issue by acoulton** [Behat/docs#180]
As mentioned in #179, it is slightly confusing that the docs version is `3.0` when in fact it will always cover the latest minor of the `3.x` version series.

Fixing this would mean:

* Adding a `3.x` branch in the repository and configuring that as the default branch
* Configuring RTD to publish a 3.x version
* Configuring RTD to redirect links to the the `3.0` version to the `3.x` version
* Disabling the `3.0` version on RTD

@stof do you have access to change the github branch settings?

I can do all the RTD things.

**Comment by stof** [comment]
@acoulton no. I don't have this kind of access. But as you are an owner of the organization (which I'm not anymore for some reason), you should have the right to do it.

**Comment by acoulton** [comment]
@stof thanks, this was on my list to come back to - Konstantin made me an owner after I added this so I now have access, just haven't had time.

**Comment by acoulton** [comment]
This is now complete, all URLs like https://docs.behat.org/en/v3.0/ now redirect to https://docs.behat.org/en/v3.x/ and the v3.0 branch is renamed to v3.x in this repo.
